protostar/migrator/maybe_tmp_directory.py

```diff
@@ -0,0 +1,36 @@
+from contextlib import AbstractContextManager
+from os import listdir, rmdir
+from pathlib import Path
+from typing import Union
+
+from protostar.protostar_exception import ProtostarException
+
+
+class MaybeTmpDirectory(AbstractContextManager):
```

Comment:
turbo nitpicky, but I find use of `@contextmanager` decorator probably would yield easier to read code here